src/components/SearchBar.jsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react'
+
+import stateData from '../utils/statedata.json'
+import useCities from './customhooks/useCities'
+import ThemeContext from '../context/ThemeContext';
+
+const SearchBar = ({setsearchData,searchData}) => {
+  const [searchText, setSearchText] = useState('')
+  const [stateName, setStateName] =  useState('Assam')
+  const [cityName, setCityName] =  useState('')
+  const cityList = useCities(stateName)
+  const searchHandler = (searchText) => 
+  { 
+    return searchData?.filter((res,i)=> res.name.toLowerCase().includes(searchText.toLowerCase())
+
+    )
+  }
+  const submitHandler = (e) => {
+    e.preventDefault() 
+    const filteredResults = searchHandler(searchText)
+    setsearchData(filteredResults)
+  }
+  return (
+    <>
+    <div>
+   
+    </div>
+     <div className='search'>
+        <form onSubmit={submitHandler}>
```

Comment:
can name handlers as `handle<something>`
React docs suggest that - `By convention, it is common to name event handlers as handle followed by the event name. You’ll often see onClick={handleClick}, onMouseEnter={handleMouseEnter}, and so on.`
refer: https://beta.reactjs.org/learn/responding-to-events#adding-event-handlers